src/block/Furnace.php

```diff
@@ -81,10 +146,121 @@ public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player
+		$tile = $this->position->getWorld()->getTile($this->position);
+		if($tile instanceof TileFurnace && !$tile->isClosed()){
+			$inventory = $tile->getInventory();
+
+			$prevCookTime = $this->cookTime;
+			$prevRemainingFuelTime = $this->remainingFuelTime;
+			$prevMaxFuelTime = $this->maxFuelTime;
+
+			$fuel = $inventory->getFuel();
+			$raw = $inventory->getSmelting();
+			$product = $inventory->getResult();
+
+			$furnaceType = $tile->getFurnaceType();
+			$smelt = $this->position->getWorld()->getServer()->getCraftingManager()->getFurnaceRecipeManager($furnaceType)->match($raw);
+			$canSmelt = ($smelt instanceof FurnaceRecipe and $raw->getCount() > 0 and (($smelt->getResult()->equals($product) and $product->getCount() < $product->getMaxStackSize()) or $product->isNull()));
+
+			if($this->remainingFuelTime <= 0 and $canSmelt and $fuel->getFuelTime() > 0 and $fuel->getCount() > 0){
+				$this->checkFuel($fuel);
+			}
+
+			if($this->remainingFuelTime > 0){
+				--$this->remainingFuelTime;
+
+				if($smelt instanceof FurnaceRecipe and $canSmelt){
+					++$this->cookTime;
+
+					if($this->cookTime >= $furnaceType->getCookDurationTicks()){
+						$product = $smelt->getResult()->setCount($product->getCount() + 1);
+
+						$ev = new FurnaceSmeltEvent($tile, $raw, $product);
+						$ev->call();
+
+						if(!$ev->isCancelled()){
+							$inventory->setResult($ev->getResult());
+							$raw->pop();
+							$inventory->setSmelting($raw);
+						}
+
+						$this->cookTime -= $furnaceType->getCookDurationTicks();
+					}
+					$this->position->getWorld()->setBlock($this->position, $this);
+				}elseif($this->remainingFuelTime <= 0){
+					$this->onStopSmelting();
+				}else{
+					$this->cookTime = 0;
+					$this->position->getWorld()->setBlock($this->position, $this);
+				}
+				if(mt_rand(1, 100) < 5){
+					$this->fireCrackle();
+				}
+				$this->position->getWorld()->scheduleDelayedBlockUpdate($this->position, 1);
+			}else{
+				$this->onStopSmelting();
+			}
+
+			$viewers = array_map(fn(Player $p) => $p->getNetworkSession()->getInvManager(), $inventory->getViewers());
+			foreach($viewers as $v){
+				if($v === null){
+					continue;
+				}
+				if($prevCookTime !== $this->cookTime){
+					$v->syncData($inventory, ContainerSetDataPacket::PROPERTY_FURNACE_SMELT_PROGRESS, $this->cookTime);
+				}
+				if($prevRemainingFuelTime !== $this->remainingFuelTime){
+					$v->syncData($inventory, ContainerSetDataPacket::PROPERTY_FURNACE_REMAINING_FUEL_TIME, $this->remainingFuelTime);
+				}
+				if($prevMaxFuelTime !== $this->maxFuelTime){
+					$v->syncData($inventory, ContainerSetDataPacket::PROPERTY_FURNACE_MAX_FUEL_TIME, $this->maxFuelTime);
+				}
+			}
+		}
+	}
+
+	public function fireCrackle() : void{
+		$tile = $this->position->getWorld()->getTile($this->position);
+		if($tile instanceof BlastFurnace){
+			$this->position->getWorld()->addSound($this->position, new BlastFurnaceFireCrackleSound);
+		}elseif ($tile instanceof Smoker) {
```

Comment:
Doesn't this violate PMMP's code style?
```suggestion
		}elseif($tile instanceof Smoker){
```